e2e/framework/framework.go

```diff
@@ -51,24 +53,42 @@ func init() {
 
 // NewDefaultFramework creates the framework with default options
 func NewDefaultFramework() (*Framework, error) {
-	e2eHome := os.Getenv("APISIX_MESH_AGENT_E2E_HOME")
 	opts := &Options{
 		Kubeconfig:        GetKubeconfig(),
 		ControlPlane:      "istio",
 		ControlPlaneImage: "localhost:5000/istio/pilot:1.9.1",
 
+		ControlPlaneDiscoveryExtraArgs: []string{},
 		ControlPlaneChartsPath: []string{
-			filepath.Join(e2eHome, "charts/istio/base"),
-			filepath.Join(e2eHome, "charts/istio/istio-discovery"),
+			filepath.Join(constant.E2eHome, "charts/istio/base"),
+			filepath.Join(constant.E2eHome, "charts/istio/istio-discovery"),
+		},
+	}
+	return NewFramework(opts)
+}
+
+// NewNacosFramework creates the framework with nacos provisioner
+func NewNacosFramework() (*Framework, error) {
+	opts := &Options{
+		Kubeconfig:        GetKubeconfig(),
+		ControlPlane:      "istio",
+		ControlPlaneImage: "localhost:5000/istio/pilot:1.9.1",
+
+		ControlPlaneDiscoveryExtraArgs: []string{
+			"--set-file", "files/injection-template.yaml=" + filepath.Join(constant.E2eHome, "charts/istio/istio-discovery-nacos/files/injection-template.yaml"),
+		},
+		ControlPlaneChartsPath: []string{
+			filepath.Join(constant.E2eHome, "charts/istio/base"),
+			filepath.Join(constant.E2eHome, "charts/istio/istio-discovery"),
 		},
 	}
 	return NewFramework(opts)
 }
 
 // NewFramework creates the framework with the given options.
 func NewFramework(opts *Options) (*Framework, error) {
-	ns := randomizeNamespace()
```

Comment:
Why change the name?

randomize is a verb.

Reply:
We could only randomize an existed thing. Here we are generating a namespace with random name.

Reply:
OK, but try not to touch other things which are not related to the topic of PR  as soon as possible in the future, we can submit other PRs to change this trivial stuff.